blueocean-dashboard/src/test/js/testResult-spec.js

```diff
@@ -1,60 +1,118 @@
-        { age: 4, name: 'failure.TestThisWillFailAbunch', duration: 0.003, errorDetails: '<some exception here>', status: 'FAILED' },
-    ];
+    class MockTestService {
+
+        constructor({regressions, existingFailed, skipped, fixed}) {
+            this.regressions = regressions || [];
+            this.existingfailed = existingFailed || [];
+            this.skipped = skipped || [];
+            this.fixed = fixed || [];
+        }
+
+        newRegressionsPager(pipeline, run) {
+            return {
+                data: this.regressions
+            };
+        }
+
+        newExistingFailedPager(pipeline, run) {
+            return {
+                data: this.existingfailed
+            };
+        }
+
+        newSkippedPager(pipeline, run) {
+            return {
+                data: this.skipped
+            };
+        }
+
+        newFixedPager(pipeline, run) {
+            return {
+                data: this.fixed
+            };
+        }
+
+        testLogs() {
+            return {
+                loadStdOut: () => {},
+                loadStdErr: () => {},
+                getStdOut: () => { return null; },
+                getStdErr: () => { return null; },
+            }
+        }
+    }
 
     it('Test fixed included', () => {
-        const wrapper = shallow(<TestResults t={t} run={bigRun} tests={bigTests} />);
+        const skipped = [
+            { age: 4, name: 'failure.TestThisWillFailAbunch', duration: 0.001, status: 'SKIPPED' },
+            { age: 4, name: 'failure.TestThisWillFailAbunch', duration: 0, status: 'SKIPPED' },
+            { age: 4, name: 'failure.TestThisWillFailAbunch', duration: 0, status: 'SKIPPED' },
+        ];
+        const fixed = [
+            { age: 0, name: 'failure.TestThisWillFailAbunch', duration: 0, status: 'PASSED', state: 'FIXED' },
+        ];
+        const existingFailed = [
+            { age: 1, name: 'failure.TestThisWillFailAbunch', duration: 0.003, errorDetails: '<some exception here>', status: 'FAILED' },
+            { age: 4, name: 'failure.TestThisWillFailAbunch', duration: 0.003, errorDetails: '<some exception here>', status: 'FAILED' },
+            { age: 4, name: 'failure.TestThisWillFailAbunch', duration: 0.003, errorDetails: '<some exception here>', status: 'FAILED' },
+        ];
+
+        const run = {
+            testSummary: {
+                existingFailed: 3,
+                failed: 0,
+                fixed: 1,
+                passed: 10,
+                regressions: 0,
+                skipped: 3,
+                total: 16,
+            },
+            '_links': {
+                tests: {
+                    href: '/pipeline/runs/1/tests'
+                }
+            }
+        };
+
+        const testService = new MockTestService({fixed: fixed, skipped: skipped, existingFailed: existingFailed});
+
+        const wrapper = mount(<TestResults t={t} run={run} testService={testService} pipeline={pipeline} />);
```

Comment:
Will this test continue to work if we use `shallow(...).html()` rather than `mount`? The latter requires some `jsdom` bootstrap to happen which would be nice to avoid. Unfortunately when we first started to use Enzyme I didn't know about the `html()` method, so some of the tests used `mount` to do deep rendering.